app/views/admin/attachments/_form.html.erb

```diff
@@ -61,13 +61,15 @@
 
   <%= hidden_field_tag :type, params[:type] %>
 
-  <% if attachment.is_a?(ExternalAttachment) %>
-    <%= render "govuk_publishing_components/components/button", {
-      text: "Next"
-    } %>
-  <% else %>
-    <%= render "govuk_publishing_components/components/button", {
-      text: "Save"
-    } %>
-  <% end %>
+  <div data-module="track-button-click" data-track-category="form-button" data-track-action=<%= "#{attachment.readable_type}-attachment-button" %>>
```

Comment:
The button component has an option to pass tracking in if that's better? https://components.publishing.service.gov.uk/component-guide/button/with_data_attributes

```
<%= render "govuk_publishing_components/components/button", {
  text: "Track this!",
  margin_bottom: true,
  data_attributes: {
    module: "track-button-click",
    "data-track-category": "form-button",
    data-track-action: "#{attachment.readable_type}-attachment-button"
  }
} %>
```

Reply:
Yes, that would be a much better use of the component. Cheers @davidgisbey. Have updated that. 👍